apps/web/pages/api/webhook/app-credential.ts

```diff
@@ -0,0 +1,76 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import z from "zod";
+
+import { appStoreMetadata } from "@calcom/app-store/appStoreMetaData";
+import { APP_CREDENTIAL_SHARING_ENABLED } from "@calcom/lib/constants";
+import prisma from "@calcom/prisma";
+
+const appCredentialWebhookRequestBodySchema = z.object({
+  // UserId of the cal.com user
+  userId: z.number().int(),
+  //   The dirname of the app under packages/app-store
+  appDirName: z.string(),
+  keys: z.record(z.any()),
+});
+/** */
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  // Check that credential sharing is enabled
+  if (!APP_CREDENTIAL_SHARING_ENABLED) {
+    return res.status(403).json({ message: "Credential sharing is not enabled" });
+  }
+
+  // Check that the webhook secret matches
+  if (req.headers["calcom-webhook-secret"] !== process.env.CALCOM_WEBHOOK_SECRET) {
+    return res.status(403).json({ message: "Invalid webhook secret" });
+  }
```

Comment:
Would this secret also be used in future webhooks as well? If we add this to the SaSS product then would need UI to generate a secret based on userId.